packages/maskbook/src/plugins/Wallet/hooks/useCollectibles.ts

```diff
@@ -28,28 +28,33 @@ export function useCollectibles(address: string, provider: CollectibleProvider,
     return useAsyncRetry(async () => {
         if (page === 1) values.current = []
         if (!address) return []
-        // a list of mock data address:
-        // 0x3c6137504c38215fea30605b3e364a23c1d3e14f
-        // 0x65c1b9ae4e4d8dcccfd3dc41b940840fe8570f2a
-        // 0xa357a589a37cf7b6edb31b707e8ed3219c8249ac
-        const result = await WalletRPC.getAssetsListNFT(address.toLowerCase(), chainId, provider, page, 50)
-        const erc721Tokens = await WalletRPC.getERC721TokensPaged(page, 50)
+        try {
+            // a list of mock data address:
+            // 0x3c6137504c38215fea30605b3e364a23c1d3e14f
+            // 0x65c1b9ae4e4d8dcccfd3dc41b940840fe8570f2a
+            // 0xa357a589a37cf7b6edb31b707e8ed3219c8249ac
+            const result = await WalletRPC.getAssetsListNFT(address.toLowerCase(), chainId, provider, page, 50)
+            const erc721Tokens = await WalletRPC.getERC721TokensPaged(page, 50)
 
-        values.current.push(
-            ...uniqWith(
-                [
-                    ...result,
-                    ...erc721Tokens.map((x) =>
-                        createERC721Token(x.chainId, x.tokenId, x.address, x.name, x.symbol, x.baseURI, x.tokenURI, {
-                            name: x.assetName,
-                            description: x.assetDescription,
-                            image: x.assetImage,
-                        }),
-                    ),
-                ],
-                (a, b) => isSameAddress(a.address, b.address) && a.tokenId === b.tokenId,
-            ),
-        )
-        return values.current
+            values.current.push(
+                ...uniqWith(
+                    [
+                        ...result,
+                        ...erc721Tokens.map((x) =>
+                            createERC721Token(x.chainId, x.tokenId, x.address, x.name, x.symbol, x.baseURI, x.tokenURI, {
+                                name: x.assetName,
+                                description: x.assetDescription,
+                                image: x.assetImage,
+                            }),
+                        ),
+                    ],
+                    (a, b) => isSameAddress(a.address, b.address) && a.tokenId === b.tokenId,
+                ),
+            )
+            return values.current
+        } catch (e) {
+            if (values.current.length === 0) throw e
+            return values.current
```

Comment:
looks strange but ok for now

Reply:
There is the hope that the previous data will be retained after the page turning request reports an error